### DISCUSSION TITLE
VEX of false-positives is apparently broken

### DISCUSSION THREAD

**Discussion by macedogm** [aquasecurity/trivy#8098] [Bugs]
### Description

Apparently the PR https://github.com/aquasecurity/trivy/pull/7985 broke the identification and suppression of false-positives with VEX.

When using a VEX report of any type in Trivy v0.57.1, the identified false-positives are properly suppressed. After updating to Trivy v0.58.0, I noticed that the false-positives are no longer being identified. I tested with VEX reports from https://github.com/rancher/vexhub/ and they work in v0.57.1, but not in v0.58.0.

I was able to pinpoint the faulty change to https://github.com/aquasecurity/trivy/pull/7985 and https://github.com/aquasecurity/trivy/blob/v0.58.0/pkg/sbom/io/encode.go, but I wasn't able yet to find the exact logic that is broken.

### Desired Behavior

False-positives from VEX reports should be suppressed.

### Actual Behavior

False-positives from VEX reports are not being identified and neither suppressed.

### Reproduction Steps

```bash
1. Use Trivy v0.58.0
2. Download the VEX report https://raw.githubusercontent.com/rancher/vexhub/refs/heads/main/pkg/golang/github.com/rancher/fleet/scan.openvex.json.
3. Execute `trivy i -s HIGH,CRITICAL --vex scan.openvex.json --show-suppressed rancher/fleet:v0.10.2`
4. See that 2 instances of CVE-2024-45337 are marked as vulnerable.
5. Repeat the same test, after clearing Trivy's cache, with Trivy v0.57.1.
6. See that the 2 instances of CVE-2024-45337 are properly marked as false-positives.

...
```

### Target

Container Image

### Scanner

Vulnerability

### Output Format

Table

### Mode

Standalone

### Debug Output

```bash
Is showed above.
```

### Operating System

OpenSUSE Slowroll

### Version

```bash
v0.58.0
```

### Checklist

- [X] Run `trivy clean --all`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by macedogm** [comment]
Sorry for the ping @knqyf263, but this might be a bad bug.

**Reply by macedogm** [reply]
Or I'm doing something really wrong 😬

**Reply by knqyf263** [reply]
Thanks. We'll take a look next week.

**Reply by knqyf263** [reply]
@DmitriyLewen I have less capacity this week. Can you please take a look?

**Reply by DmitriyLewen** [reply]
@macedogm thanks for your report!
This is bug. I created #8102

I will work on this today.

**Reply by macedogm** [reply]
Thanks Dmitriy and Teppei for the quick solution.

**Comment by DmitriyLewen** [comment]
Created #8102
